src/Microsoft.Android.Build.BaseTasks/AndroidAsyncTask.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using Xamarin.Build;
+using static System.Threading.Tasks.TaskExtensions;
+
+namespace Microsoft.Android.Build.Tasks
+{
+	public abstract class AndroidAsyncTask : AsyncTask
+	{
+		/// <summary>
+		/// A helper for non-async overrides of RunTaskAsync, etc.
+		/// </summary>
+		public static readonly System.Threading.Tasks.Task Done =
+			System.Threading.Tasks.Task.CompletedTask;
+
+		public abstract string TaskPrefix { get; }
+
+		public override bool Execute ()
+		{
+			try {
+				return RunTask ();
+			} catch (Exception ex) {
+				this.LogUnhandledException (TaskPrefix, ex);
+				return false;
+			}
+		}
+
+		public virtual bool RunTask ()
```

Comment:
Should this actually be `virtual`, and not *non*-virtual?

Reply:
I added a comment for this method.